lib/iiif_print.rb

```diff
@@ -89,4 +90,43 @@ def iiif_print_config?
       end
     end
   end
+
+  # @api public
+  #
+  # Map the given model's metadata to the given IIIF version spec's metadata structure.  This
+  # is intended to be a drop-in replacement for `Hyrax::IiifManifestPresenter#manifest_metadata`.
+  #
+  # @param model [Object]
+  # @param version [Integer]
+  # @param fields [Array<IiifPrint::Metadata::Field>, Array<#name, #label>]
+  # @return [Array<Hash>]
+  #
+  # @see specs for expected output
+  #
+  # @see Hyrax::IiifManifestPresenter#manifest_metadata
+  def self.manifest_metadata_for(model:,
+                                 version: config.default_iiif_manifest_version,
+                                 fields: default_fields_for(model))
+    Metadata.manifest_for(model: model, version: version, fields: fields)
+  end
+
+  def self.metadata_fields
+    config.metadata_fields
+  end
+
+  # Hash is an arbitrary attribute key/value pairs
+  # Struct is a defined set of attribute "keys".  When we favor defined values,
+  # then we are naming the concept and defining the range of potential values.
+  Field = Struct.new(:name, :label, :options, keyword_init: true)
+
+  # TODO: figure out a way to use a custom label, right now it takes it get rendered from the title
+  def self.default_fields_for(_model, fields: metadata_fields)
```

Comment:
With the TODO item, let's mark this as `@api private`; we haven't settled on the structure.